src/core/AuthKitCore.ts

```diff
@@ -0,0 +1,241 @@
+      return false;
+    }
+
+    const currentTime = Math.floor(Date.now() / 1000);
+    return expiryTime - currentTime <= buffer;
+  }
+
+  /**
+   * Get the expiry time from a token's claims.
+   *
+   * @param token - The JWT access token
+   * @returns Unix timestamp of expiry, or null if not present
+   */
+  private getTokenExpiryTime(token: string): number | null {
+    const claims = this.parseTokenClaims(token);
+    return claims.exp ?? null;
+  }
+
+  /**
+   * Parse JWT claims from an access token.
+   *
+   * @param token - The JWT access token
+   * @returns Decoded token claims
+   * @throws Error if token is invalid
+   */
+  parseTokenClaims<TCustomClaims = CustomClaims>(
+    token: string,
+  ): BaseTokenClaims & TCustomClaims {
+    try {
+      return decodeJwt<BaseTokenClaims & TCustomClaims>(token);
+    } catch (error) {
+      throw new Error('Invalid token');
+    }
+  }
+
+  /**
+   * Encrypt a session object into a string suitable for cookie storage.
+   *
+   * @param session - The session to encrypt
+   * @returns Encrypted session string
+   * @throws SessionEncryptionError if encryption fails
+   */
+  async encryptSession(session: Session): Promise<string> {
+    try {
+      const password = this.config.getValue('cookiePassword');
+      const encryptedSession = await this.encryption.sealData(session, {
+        password,
+        ttl: 0,
+      });
+      return encryptedSession;
+    } catch (error) {
+      throw new SessionEncryptionError('Failed to encrypt session', error);
+    }
+  }
+
+  /**
+   * Decrypt an encrypted session string back into a session object.
+   *
+   * @param encryptedSession - The encrypted session string
+   * @returns Decrypted session object
+   * @throws SessionEncryptionError if decryption fails
+   */
+  async decryptSession(encryptedSession: string): Promise<Session> {
+    try {
+      const password = this.config.getValue('cookiePassword');
+      const session = await this.encryption.unsealData<Session>(
+        encryptedSession,
+        { password },
+      );
+      return session;
+    } catch (error) {
+      throw new SessionEncryptionError('Failed to decrypt session', error);
+    }
+  }
+
+  /**
+   * Refresh tokens using WorkOS API.
+   *
+   * @param refreshToken - The refresh token
+   * @param organizationId - Optional organization ID to switch to
+   * @returns New access token, refresh token, user, and impersonator
+   * @throws TokenRefreshError if refresh fails
+   */
+  async refreshTokens(
+    refreshToken: string,
+    organizationId?: string,
+  ): Promise<{
+    accessToken: string;
+    refreshToken: string;
+    user: User;
+    impersonator: Impersonator | undefined;
+  }> {
+    try {
+      const result =
+        await this.client.userManagement.authenticateWithRefreshToken({
+          refreshToken,
+          clientId: this.clientId,
+          organizationId,
+        });
+
+      return {
+        accessToken: result.accessToken,
```

Comment:
another place where i would expect to see an organizationId (or if we're not using the top-level organizationId i think we could probably also ditch top-level impersonator).